### PR TITLE
Add multiple detectors with different parameters for each

### PR DESCRIPTION
**Pull request by ghazalemir**
The previous version was considering same parameters for all the detectors, this new version consider different parameters for each individual detector.

### CODE REVIEW

File: source/rpt/parameters_rpt.cc [hunk 1]
```diff
@@ -202,7 +200,56 @@ Parameters::DetectorParameters::parse_parameters(ParameterHandler &prm)
     radius                  = prm.get_double("radius");
     length                  = prm.get_double("length");
     detector_positions_file = prm.get("detector positions file");
+
```
Comment (acdaigneault) [inline source/rpt/parameters_rpt.cc:203]: I don't know if this part needs more comments. Looks good to me.

File: Testing/Temporary/LastTest.log [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Start testing: Oct 18 11:38 EDT
```
Comment (acdaigneault) [inline Testing/Temporary/LastTest.log:1]: Is this file necessary?

File: tests/rpt/Testing/Temporary/CTestCostData.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+---
```
Comment (acdaigneault) [inline tests/rpt/Testing/Temporary/CTestCostData.txt:1]: Temp file?

File: tests/rpt/Testing/Temporary/LastTest.log [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Start testing: Oct 18 09:37 EDT
```
Comment (acdaigneault) [inline tests/rpt/Testing/Temporary/LastTest.log:1]: Temp file?

File: include/rpt/parameters_rpt.h [hunk 5]
```diff
@@ -69,13 +69,9 @@ namespace Parameters
     double peak_to_total_ratio;
     double sampling_time; // [s]
 
-    // Parameters to tune or fixed parameters if tuning is disable
-    double dead_time;          // Dead time of the detector per accepted pulse
-    double activity;           // Activity of the tracer
-    double gamma_rays_emitted; // Number of gamma-rays emitted by each
+    // Fixed parameters
+      double gamma_rays_emitted; // Number of gamma-rays emitted by each
```
Comment (blaisb) [inline include/rpt/parameters_rpt.h:73]: You need to apply automatic indentation. you can run /contrib/utilities/indent_all.sh . Ask Toni, Lucka, Shahab or Olivier if you need help :)!

File: include/rpt/parameters_rpt.h [hunk 6]
```diff
@@ -126,6 +122,12 @@ namespace Parameters
     double      length;
     std::string detector_positions_file;
 
+    // Tuned parameters
+    std::vector<double> dead_time;          // Dead time of the detector per accepted pulse
+    std::vector<double> activity;           // Activity of the tracer
+    std::vector<double> attenuation_coefficient_reactor; // Total linear attenuation
+    // coefficient of the medium
```
Comment (blaisb) [inline include/rpt/parameters_rpt.h:129]:
```suggestion
    //Total linear attenuation coefficient of the medium
    std::vector<double> attenuation_coefficient_reactor; 
```

File: include/rpt/particle_detector_interactions.h [hunk 7]
```diff
@@ -57,6 +57,9 @@ class ParticleDetectorInteractions
     , detector_radius(detector.get_radius())
     , detector_length(detector.get_length())
     , parameters(rpt_parameters)
+    , dead_time(detector.dead_time)
```
Comment (blaisb) [inline include/rpt/particle_detector_interactions.h:60]: Add the description of these arguments after line 49 above. Every parameter must be detailed :)!

File: source/rpt/parameters_rpt.cc [hunk 8]
```diff
@@ -190,6 +173,21 @@ Parameters::DetectorParameters::declare_parameters(ParameterHandler &prm)
                       "none",
                       Patterns::FileName(),
                       "Detector positions file name");
+
+    prm.declare_entry("dead time",
+                      "1",
+                      Patterns::Anything(),
```
Comment (blaisb) [inline source/rpt/parameters_rpt.cc:179]:
```suggestion
                                                Patterns::List(Patterns::Double()),
```
The right pattern is a list of doubles and not anything. So you can actually specify that what you want and need is a list of doubles :)
See this part in Lethe for more details : https://github.com/lethe-cfd/lethe/blob/cda6095c0c3f2df9c2e0fb9e1dfca967d350536d/source/core/parameters_lagrangian.cc#L318

File: source/rpt/parameters_rpt.cc (hunk 1 above)
Comment (blaisb) [reply source/rpt/parameters_rpt.cc:203]: I would maybe add a quick comment here on the format of the file.
Like just quickly recap in a comment what are the headers to be expected. The more this is documented, the better it is.

File: source/rpt/parameters_rpt.cc [hunk 9]
```diff
@@ -202,7 +200,56 @@ Parameters::DetectorParameters::parse_parameters(ParameterHandler &prm)
     radius                  = prm.get_double("radius");
     length                  = prm.get_double("length");
     detector_positions_file = prm.get("detector positions file");
+
+    // Transform string to vector of double
+    std::string dead_time_string  = prm.get("dead time");
```
Comment (blaisb) [inline source/rpt/parameters_rpt.cc:205]: See how this is done here : 
https://github.com/lethe-cfd/lethe/blob/cda6095c0c3f2df9c2e0fb9e1dfca967d350536d/source/core/parameters_lagrangian.cc#L363

Use the same approach, it will make the code simpler and less error prone (and also more readable :))

File: tests/rpt/Testing/Temporary/CTestCostData.txt (hunk 3 above)
Comment (blaisb) [reply tests/rpt/Testing/Temporary/CTestCostData.txt:1]: to remove

File: tests/rpt/Testing/Temporary/LastTest.log (hunk 4 above)
Comment (blaisb) [reply tests/rpt/Testing/Temporary/LastTest.log:1]: To remove

**Review by blaisb — APPROVED** [review]
Good for merging. Bravo @ghazalemir  your first full PR alone :)
